File: 图灵/爬虫15期进阶/逆向课/day7-算法逆向/123.py
import hashlib
import time

import requests

def sha1_test2(arg):
    sha1 = hashlib.sha256()
    sha1.update(arg.encode('utf-8'))
    return sha1.hexdigest()


headers = {
    "accept": "*/*",
    "accept-language": "zh-CN,zh;q=0.9,en;q=0.8",
    "cache-control": "no-cache",
    "pragma": "no-cache",
    "priority": "u=1, i",
    "referer": "https://tousu.sina.com.cn/",
    "sec-ch-ua": "\"Not/A)Brand\";v=\"8\", \"Chromium\";v=\"126\", \"Google Chrome\";v=\"126\"",
    "sec-ch-ua-mobile": "?0",
    "sec-ch-ua-platform": "\"macOS\"",
    "sec-fetch-dest": "empty",
    "sec-fetch-mode": "cors",
    "sec-fetch-site": "same-origin",
    "user-agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36",
    "x-requested-with": "XMLHttpRequest"
}
cookies = {
    "UOR": ",tousu.sina.com.cn,",
    "SINAGLOBAL": "117.147.9.165_1719724809.791840",
    "ULV": "1719789806039:3:1:3::1719725409075",
    "Apache": "117.147.9.165_1719789806.274460"
}
url = "https://tousu.sina.com.cn/api/index/feed"
_ts = str(int(time.time() * 1000))

params = [
    "$d6eb7ff91ee257475%",
    "1",
    "1",
    "10",
    _ts,
    "hOPUNHqaz2dxUISk"
]
env_par = "".join(params)

sign =sha1_test2(env_par)
print(sign)
params = {
    "ts": _ts,
    "rs": "hOPUNHqaz2dxUISk", # 随机 写死
    "signature": sign,
    "type": "1",
    "page_size": "10",
    "page": "1"
}

response = requests.get(url, headers=headers, cookies=cookies, params=params)

print(response.text)
print(response)

File: 图灵/爬虫15期进阶/逆向课/day7-算法逆向/123.js
// 引用 crypto-js 加密模块
var CryptoJS = require('crypto-js')

function HMACEncrypt() {
    var text = "1"
    var key = "secret"   // 密钥文件  前端会出现  逆向的时候碰到hmac算法，需要有一个key
    return CryptoJS.HmacMD5(text, key).toString();
    // return CryptoJS.HmacSHA1(text, key).toString();
    // return CryptoJS.HmacSHA256(text, key).toString();
}
console.log(HMACEncrypt(), HMACEncrypt().length)
function SHA1Encrypt(arg) {
    return CryptoJS.SHA512(arg).toString();
}

h = function (e, t, r) {
    var n = ""
        , i = t
        ,
        a = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z", "A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z"];
    e && (i = Math.round(Math.random() * (r - t)) + t);
    for (var o = 0; o < i; o++) {
        n += a[Math.round(Math.random() * (a.length - 1))]
    }
    return n
}(!1, 16)
u = "$d6eb7ff91ee257475%"
l = 10

d = new Date().getTime()


function getSign(page, type) {
    params = [
    u,
    page, // page
    "10", // page_size 写死
    d,
    type, // type // 写活
    h
]
    para = {}
    sign = SHA1Encrypt(params.join(""))
    return para["ts"] = d,
        para['rs'] = h,
        para['signature'] = sign,
        para['page_size'] = "10",
        para['type'] = type,
        para['page'] = page,
        para

}
